packages/data-stores/src/mutations/use-domains-bulk-actions-mutation.ts

```diff
@@ -49,12 +54,27 @@ export function useDomainsBulkActionsMutation<
 					} );
 			}
 		},
+		onSuccess: ( data: TData, variables: BulkUpdateVariables ) => {
+			const { type } = variables;
+
+			if ( type !== 'set-auto-renew' ) {
+				return;
+			}
+
+			const { blogIds } = variables;
+
+			// Refreshes data for all the domains that were updated
+			blogIds.forEach( ( blogId ) => {
+				queryClient.invalidateQueries( { queryKey: getSiteDomainsQueryKey( blogId ) } );
+				queryClient.invalidateQueries( { queryKey: getAllDomainsQueryKey( { no_wpcom: true } ) } );
+			} );
+		},
```

Comment:
```suggestion
		onSuccess: ( data, variables ) => {
			if ( variables.type !== 'set-auto-renew' ) {
				return;
			}

			queryClient.invalidateQueries( { queryKey: getBulkDomainUpdateStatusQueryKey() } );

			// Refreshes data for all the domains that were updated
			variables.blogIds.forEach( ( blogId ) => {
				queryClient.invalidateQueries( { queryKey: getSiteDomainsQueryKey( blogId ) } );
				queryClient.invalidateQueries( { queryKey: getAllDomainsQueryKey( { no_wpcom: true } ) } );
			} );
		},
```
I have two main points here:
1. We should invalidate the `bulkDomainUpdateStatusQuery`. This is what will reliably display the success notice.
2. We don't need to specify the types for `data` and `context`. `react-query` really shines in terms of TS support, so we already get those type definitions for free.

My other remarks about object destructuring are optional, feel free to disregard if you prefer.